robot_sf/render/sim_view.py

```
@@ -181,12 +213,22 @@
     def exit(self):
+        """Exit the simulation."""
+        logger.debug("Exiting the simulation.")
         self.is_exit_requested = True
+        self._handle_quit()
 
     def _handle_quit(self, e=None):
         """Handle the quit event of the pygame window."""
         self.is_exit_requested = True
         self.is_abortion_requested = True
+        if self.record_video and self.frames:
+            if MOVIEPY_AVAILABLE:
```

Ruff

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)